From 7fa5c4fe97cf27dae4bee3f0d1b96ca1940f6f68 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Fri, 18 Jul 2014 03:10:12 +1000
Subject: [PATCH 2/3] mythbackend: fixes QUERY_FILE_HASH hanging

The frontend will typically call the backend with an identifier, and not its IP address. As such, the backend attempt to transfer the QUERY_FILE_HASH command to anoyher backend (which is itself) resulting in an infinite loop

Fixes #12212
---
 mythtv/programs/mythbackend/mainserver.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythbackend/mainserver.cpp b/mythtv/programs/mythbackend/mainserver.cpp
index 5160a13..abf67c6 100644
--- a/mythtv/programs/mythbackend/mainserver.cpp
+++ b/mythtv/programs/mythbackend/mainserver.cpp
@@ -3253,7 +3253,7 @@ void MainServer::HandleQueryFileHash(QStringList &slist, PlaybackSock *pbs)
 
     QString hash = "";
 
-    if (hostname == gCoreContext->GetHostName())
+    if (gCoreContext->IsThisHost(hostname))
     {
         StorageGroup sgroup(storageGroup, gCoreContext->GetHostName());
         QString fullname = sgroup.FindFile(filename);
-- 
1.7.10.2

